netlogo-gui/src/main/editor/EditorMenu.scala

```diff
@@ -0,0 +1,9 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.editor
+
+import javax.swing.Action
+
+trait EditorMenu {
+  def offerAction(action: Action): Unit
```

Comment:
Wow!, that's very smart. I think this approach can be used in a lot of Swing apps.